src/agent/communicator/src/https_socket.cpp

```diff
@@ -0,0 +1,257 @@
+#include "https_socket.hpp"
+
+#include "http_client_utils.hpp"
+#include <ihttp_socket.hpp>
+#include <logger.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/asio/ssl.hpp>
+#include <boost/beast.hpp>
+#include <boost/system/error_code.hpp>
+
+#include <exception>
+#include <string>
+
+namespace http_client
+{
+    void HttpsSocket::Connect(boost::asio::io_context& ioContext,
+                              const boost::asio::ip::tcp::resolver::results_type& endpoints,
+                              boost::system::error_code& ec,
+                              const std::chrono::seconds timeOut)
+    {
+        try
+        {
+            bool connectionSuccess = false;
+
+            // Start the asynchronous connect operation
+            boost::asio::async_connect(m_ssl_socket.lowest_layer(),
+                                       endpoints,
+                                       [&](const boost::system::error_code& errorCode, const auto&)
+                                       {
+                                           if (!errorCode)
+                                           {
+                                               connectionSuccess = true;
+                                               ec = errorCode;
+                                           }
+                                       });
+
+            ioContext.run_for(timeOut); // Run for 2 seconds();
+
+            if (connectionSuccess)
+            {
+                m_ssl_socket.handshake(boost::asio::ssl::stream_base::client, ec); // NOLINT
```

Comment:
Which warning is being silenced? Can we put it in the `NOLINT` comment?

Reply:
Done